common/shared.c

```diff
@@ -86,6 +86,7 @@ static int getaddr(char *node, char *service,
 {
 	int ret;
 	struct fi_info *fi;
+	struct fi_info hcopy = *hints;
```

Comment:
Will fi_getinfo() guarantee to not modify the hints?  If so, we can just pass it in directly.